x-pack/plugins/observability_solution/observability/public/pages/alert_details/alert_details_contextual_insights.tsx

```diff
@@ -28,40 +28,42 @@ export function AlertDetailContextualInsights({ alert }: { alert: AlertData | nu
       return [];
     }
 
-    const res = await http.get('/internal/apm/assistant/get_obs_alert_details_context', {
-      query: {
-        alert_started_at: new Date(alert.formatted.start).toISOString(),
-
-        // service fields
-        'service.name': fields['service.name'],
-        'service.environment': fields['service.environment'],
-        'transaction.type': fields['transaction.type'],
-        'transaction.name': fields['transaction.name'],
-
-        // infra fields
-        'host.name': fields['host.name'],
-        'container.id': fields['container.id'],
-      },
-    });
-
-    const {
-      serviceSummary,
-      downstreamDependencies,
-      logCategories,
-      serviceChangePoints,
-      exitSpanChangePoints,
-      anomalies,
-    } = res as any;
-
-    const serviceName = fields['service.name'];
-    const serviceEnvironment = fields['service.environment'];
-
-    const obsAlertContext = `${
-      !isEmpty(serviceSummary)
-        ? `Metadata for the service where the alert occurred:
+    try {
+      const res = await http.get('/internal/apm/assistant/get_obs_alert_details_context', {
```

Comment:
I think this is not the recommended way of doing it, right? AFAICT we should not call solution's APIs from other places. Can you move this to `apm_data_access`?

Reply:
This will eventually be moved to the observability plugin since this is supposed to return multiple signals (not just APM). However this is a bigger change. I can't move this to `apm_data_access` because it would create a dependency on the apm plugin, and will cause a circular dependency (`apm` already depends on `apm_data_access`).